File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Hbr/AlibabaCloud/SDK/ROS/CDK/Hbr/DbPlanProps.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Hbr
{
    #pragma warning disable CS8618

    /// <summary>Properties for defining a `ALIYUN::HBR::DbPlan`.</summary>
    [JsiiByValue(fqn: "@alicloud/ros-cdk-hbr.DbPlanProps")]
    public class DbPlanProps : AlibabaCloud.SDK.ROS.CDK.Hbr.IDbPlanProps
    {
        /// <summary>Property dbPlanName: Display name of the backup plan.</summary>
        [JsiiProperty(name: "dbPlanName", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        public object DbPlanName
        {
            get;
            set;
        }

        /// <summary>Property hostUuid: Uuid of the host of the database instance.</summary>
        [JsiiProperty(name: "hostUuid", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        public object HostUuid
        {
            get;
            set;
        }

        /// <summary>Property sourceType: Database type, allowed value: MYSQL, ORACLE, MSSQL.</summary>
        [JsiiProperty(name: "sourceType", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        public object SourceType
        {
            get;
            set;
        }

        /// <summary>Property vaultId: Vault ID to create backup plan, the backup data will be stored to the vault.</summary>
        [JsiiProperty(name: "vaultId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        public object VaultId
        {
            get;
            set;
        }

        /// <summary>Property continuousPlan: Continuous backup plan schedule.</summary>
        /// <remarks>
        /// Use {   "type": "continuous" }.
        /// </remarks>
        [JsiiOptional]
        [JsiiProperty(name: "continuousPlan", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOptional: true)]
        public object? ContinuousPlan
        {
            get;
            set;
        }

        /// <summary>Property cumulativePlan: Cumulative plan schedule, only for mssql.</summary>
        /// <remarks>
        /// More details see FullPlan.
        /// </remarks>
        [JsiiOptional]
        [JsiiProperty(name: "cumulativePlan", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOptional: true)]
        public object? CumulativePlan
        {
            get;
            set;
        }

        /// <summary>Property fullPlan: Full backup plan schedule.</summary>
        /// <remarks>
        /// daily: {"type": "daily", "start": "00:00:00", "interval": 3}, weekly {"type":"weekly","start": "03:00:00","days": [1,2,3,4,5],"interval": 1}, days can be 0 - 6, 0 means Sunday, and interval can be 1 - 52.
        /// </remarks>
        [JsiiOptional]
        [JsiiProperty(name: "fullPlan", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOptional: true)]
        public object? FullPlan
        {
            get;
            set;
        }

        /// <summary>Property incPlan: Incremental backup plan schedule.</summary>
        /// <remarks>
        /// Only for mysql and oracle. More details see FullPlan.
        /// </remarks>
        [JsiiOptional]
        [JsiiProperty(name: "incPlan", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOptional: true)]
        public object? IncPlan
        {
            get;
            set;
        }

        /// <summary>Property instanceUuid: Uuid of database instance.</summary>
        [JsiiOptional]
        [JsiiProperty(name: "instanceUuid", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOptional: true)]
        public object? InstanceUuid
        {
            get;
            set;
        }

        /// <summary>Property logPlan: Log backup plan schedule.More details see FullPlan.</summary>
        [JsiiOptional]
        [JsiiProperty(name: "logPlan", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOptional: true)]
        public object? LogPlan
        {
            get;
            set;
        }

        /// <summary>Property maxRateLimit: Max rate limit for backup job,.</summary>
        [JsiiOptional]
        [JsiiProperty(name: "maxRateLimit", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"number\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOptional: true)]
        public object? MaxRateLimit
        {
            get;
            set;
        }

        /// <summary>Property maxRetrySeconds: Max retry seconds on network failure.</summary>
        [JsiiOptional]
        [JsiiProperty(name: "maxRetrySeconds", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"number\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOptional: true)]
        public object? MaxRetrySeconds
        {
            get;
            set;
        }

        /// <summary>Property options: Backup options in json format, different for each type of database.</summary>
        /// <remarks>
        /// For Oracle, use {"channels":4,"compression":"lzop","offline_backup":false,"archivelog_reserve_hours":24,"custom_commands":""}, "channels" means numbers of concurrent theads, "archivelog_reserve_hours" means how long before the archive log will be deleted after backup job completed, other paramters should use the default vaule. For Mysql, use {"channels":4,"compression":"lzop","del_binlog":false}, "del_binlog" means whether the binlog will be deleted after backup completed, only take effect for log or continuous backup. For SQL Server, use {"channels":4,"verify":false,"compression":"lzop","backup_new_databases":false}.
        /// </remarks>
        [JsiiOptional]
        [JsiiProperty(name: "options", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOptional: true)]
        public object? Options
        {
            get;
            set;
        }

        /// <summary>Property source: Which database instance or database will be backup.</summary>
        [JsiiOptional]
        [JsiiProperty(name: "source", typeJson: "{\"union\":{\"types\":[{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"},{\"fqn\":\"@alicloud/ros-cdk-hbr.RosDbPlan.SourceProperty\"}]}}", isOptional: true)]
        public object? Source
        {
            get;
            set;
        }
    }
}

File: multiple-languages/go/alicloudroscdkhbr/DbPlanProps.go
package alicloudroscdkhbr


// Properties for defining a `DbPlan`.
//
// See https://www.alibabacloud.com/help/ros/developer-reference/aliyun-hbr-dbplan
type DbPlanProps struct {
	// Property dbPlanName: Display name of the backup plan.
	DbPlanName interface{} `field:"required" json:"dbPlanName" yaml:"dbPlanName"`
	// Property hostUuid: Uuid of the host of the database instance.
	HostUuid interface{} `field:"required" json:"hostUuid" yaml:"hostUuid"`
	// Property sourceType: Database type, allowed value: MYSQL, ORACLE, MSSQL.
	SourceType interface{} `field:"required" json:"sourceType" yaml:"sourceType"`
	// Property vaultId: Vault ID to create backup plan, the backup data will be stored to the vault.
	VaultId interface{} `field:"required" json:"vaultId" yaml:"vaultId"`
	// Property continuousPlan: Continuous backup plan schedule.
	//
	// Use {   "type": "continuous" }.
	ContinuousPlan interface{} `field:"optional" json:"continuousPlan" yaml:"continuousPlan"`
	// Property cumulativePlan: Cumulative plan schedule, only for mssql.
	//
	// More details see FullPlan.
	CumulativePlan interface{} `field:"optional" json:"cumulativePlan" yaml:"cumulativePlan"`
	// Property fullPlan: Full backup plan schedule.
	//
	// daily: {"type": "daily", "start": "00:00:00", "interval": 3}, weekly {"type":"weekly","start": "03:00:00","days": [1,2,3,4,5],"interval": 1}, days can be 0 - 6, 0 means Sunday, and interval can be 1 - 52.
	FullPlan interface{} `field:"optional" json:"fullPlan" yaml:"fullPlan"`
	// Property incPlan: Incremental backup plan schedule.
	//
	// Only for mysql and oracle. More details see FullPlan.
	IncPlan interface{} `field:"optional" json:"incPlan" yaml:"incPlan"`
	// Property instanceUuid: Uuid of database instance.
	InstanceUuid interface{} `field:"optional" json:"instanceUuid" yaml:"instanceUuid"`
	// Property logPlan: Log backup plan schedule.More details see FullPlan.
	LogPlan interface{} `field:"optional" json:"logPlan" yaml:"logPlan"`
	// Property maxRateLimit: Max rate limit for backup job,.
	MaxRateLimit interface{} `field:"optional" json:"maxRateLimit" yaml:"maxRateLimit"`
	// Property maxRetrySeconds: Max retry seconds on network failure.
	MaxRetrySeconds interface{} `field:"optional" json:"maxRetrySeconds" yaml:"maxRetrySeconds"`
	// Property options: Backup options in json format, different for each type of database.
	//
	// For Oracle, use {"channels":4,"compression":"lzop","offline_backup":false,"archivelog_reserve_hours":24,"custom_commands":""}, "channels" means numbers of concurrent theads, "archivelog_reserve_hours" means how long before the archive log will be deleted after backup job completed, other paramters should use the default vaule. For Mysql, use {"channels":4,"compression":"lzop","del_binlog":false}, "del_binlog" means whether the binlog will be deleted after backup completed, only take effect for log or continuous backup. For SQL Server, use {\"channels\":4,\"verify\":false,\"compression\":\"lzop\",\"backup_new_databases\":false}.
	Options interface{} `field:"optional" json:"options" yaml:"options"`
	// Property source: Which database instance or database will be backup.
	Source interface{} `field:"optional" json:"source" yaml:"source"`
}

